File: benchmarks/solutions/c/a1-two-sum.c
#include <stdlib.h>

struct Result { int i; int j; };

struct Result two_sum(int *nums, int len, int target) {
    for (int i = 0; i < len; i++) {
        for (int j = i + 1; j < len; j++) {
            if (nums[i] + nums[j] == target) {
                return (struct Result){i, j};
            }
        }
    }
    return (struct Result){-1, -1};
}

File: benchmarks/solutions/elixir/a1-two-sum.exs
defmodule TwoSum do
  def two_sum(nums, target) do
    nums
    |> Enum.with_index()
    |> Enum.reduce_while(%{}, fn {n, i}, seen ->
      complement = target - n
      case Map.fetch(seen, complement) do
        {:ok, j} -> {:halt, {j, i}}
        :error -> {:cont, Map.put(seen, n, i)}
      end
    end)
  end
end
